src/simulations/Verlet.cpp

```diff
@@ -1,75 +1,86 @@
 #include "Verlet.h"
-
 #include "io/output/WriterFactory.h"
+#include "strategies/StrategyFactory.h"
 #include "utils/Arguments.h"
 #include "utils/ArrayUtils.h"
 #include "utils/CLIUtils.h"
+#include "utils/StringUtils.h"
+#include <functional>
+#include <memory>
+#include <spdlog/spdlog.h>
+#include <string>
 
-Verlet::Verlet(const ParticleContainer &pc, const Arguments &args)
+Verlet::Verlet(const ParticleContainer &pc, const Arguments &args, int type)
     : m_particles{pc}, m_startTime{args.startTime}, m_endTime{args.endTime}, m_delta_t{args.delta_t},
-      m_itFreq{args.itFreq}, m_type{args.type} {};
-
-Verlet::Verlet(const std::string &filename, const Arguments &args)
+      m_itFreq{args.itFreq}, m_type{args.type} {
+    SPDLOG_TRACE("Created Verlet simulation with ParticleContainer {} and Arguments {}", pc.toString(),
+                 args.toString());
+    initializeSimulation(type);
+}
+Verlet::Verlet(const std::string &filename, const Arguments &args, int type)
     : m_startTime{args.startTime}, m_endTime{args.endTime}, m_delta_t{args.delta_t}, m_itFreq{args.itFreq},
       m_type{args.type} {
     m_particles.fromFile(filename);
-};
-
-Verlet::Verlet(const Arguments &args)
+    SPDLOG_TRACE("Created Verlet simulation from file {} with Arguments {}", filename, args.toString());
+    initializeSimulation(type);
+}
+Verlet::Verlet(const Arguments &args, int type)
     : m_startTime{args.startTime}, m_endTime{args.endTime}, m_delta_t{args.delta_t}, m_itFreq{args.itFreq},
-      m_type{args.type} {};
-Verlet::~Verlet() = default;
+      m_type{args.type} {
+    SPDLOG_TRACE("Created Verlet simulation with Arguments {}", args.toString());
+    initializeSimulation(type);
+}
+Verlet::~Verlet() { SPDLOG_TRACE("Destroyed Verlet object."); }
+
+void Verlet::initializeSimulation(int type) {
+    SPDLOG_TRACE("Initializing Verlet simulation...");
+
+    StrategyFactory sf;
+    m_writer = WriterFactory::createWriter(m_type);
+    auto [cv, cx, cf] = StrategyFactory::getSimulationFunctions(SimulationType::VERLET, type);
```

Comment:
Regarding the different classes: We thought that each simulation would be different enough from eachother to warrant separate files, but yeah, turns out they're much more similar than we initially thought (for now). I'll currently leave them be to avoid headaches, since we already started working on Assignment 3  and I don't want the merge process to potentially be painful, but we'll consider refactoring this part to avoid repetition.

Regarding "Verlet" vs. "Gravity": fair point, renamed it to "Gravity" throughout.